great_expectations/datasource/fluent/interfaces.py

```diff
@@ -95,6 +96,13 @@
     )
 
 
+class _PartitionerProtocol(Protocol):
```

Comment:
There is a `_Partitioner` protocol in both `sql_datasource.py` and `spark_generic_partitioners.py` which are almost identical, the sql one has 1 extra method. It seems like both those protocols should extend this one. Could you do that in this PR?
It also seems like we should pull more up to this parent protocol (eg everything but `param_defaults`). Looking at the code and docstrings, it looks to have been a historical copy/paste.

Reply:
Done! Good call